x-pack/plugins/security_solution/server/endpoint/services/feature_usage/service.ts

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { LicensingPluginSetup, LicensingPluginStart } from '../../../../../licensing/server';
+
+const FEATURES = Object.freeze({
+  HOST_ISOLATION: 'Host isolation',
+  HOST_ISOLATION_EXCEPTION: 'Host isolation exception',
+  HOST_ISOLATION_EXCEPTION_BY_POLICY: 'Host isolation exception by policy',
+  TRUST_APP_BY_POLICY: 'Trusted app by policy',
+  EVENT_FILTERS_BY_POLICY: 'Event filters by policy',
+  RANSOMWARE_PROTECTION: 'Ransomeware protection',
+  MEMORY_THREAT_PROTECTION: 'Memory threat protection',
+  BEHAVIOR_PROTECTION: 'Behavior protection',
+});
+
+export class FeatureUsageService {
+  private licensingPluginStart: LicensingPluginStart | undefined;
+
+  private get notify(): (featureName: string, usedAt?: number | Date | undefined) => void {
```

Comment:
There maybe a better way to define the type for `featureName` by creating a Union from the object's values above. I was trying to see if there as a Uiltiy for that in noe of our current kibana packages (lookat at `@kbn/utility-types` and maybe a combination of `Value<>` and somethign else? not sure), but did not see anything for it right a way.

Reply:
I'll take another look. I'll remove the `usedAt` too, we're not using it here.